pdp/flags_mapper_pca_test.go

```diff
@@ -94,3 +97,58 @@ func TestFlagsMapperPCAOrdering(t *testing.T) {
 		}
 	}
 }
+
+const (
+	expectEmptyFlagMapperPCAJSON = `{"type":"flagsMapperPCA","def":"\"\"","err":"\"\"","alg":{"type":"firstApplicableEffectPCA"}}`
+	expectFlagMapperPCAJSON      = `{"type":"flagsMapperPCA","def":"\"first\"","err":"\"second\"","alg":{"type":"firstApplicableEffectPCA"}}`
+)
+
+func TestFlagsMapperPCAMarshal(t *testing.T) {
+	ft, err := NewFlagsType("flags", "third", "first", "second")
+	if err != nil {
+		t.Fatalf("Expected no error but got: %s", err)
+	}
+	policies := []Evaluable{
+		makeSimplePermitPolicyWithObligations(
+			"first",
+			makeSingleStringObligation("order", "first"),
+		),
+		makeSimplePermitPolicyWithObligations(
+			"second",
+			makeSingleStringObligation("order", "second"),
+		),
+		makeSimplePermitPolicyWithObligations(
+			"third",
+			makeSingleStringObligation("order", "third"),
+		),
+	}
+	algParam := MapperPCAParams{
+		Argument:  AttributeDesignator{a: Attribute{id: "f", t: ft}},
+		Order:     MapperPCAInternalOrder,
+		Algorithm: firstApplicableEffectPCA{},
+	}
+	alg := makeMapperPCA(policies, algParam)
+	b, err := json.Marshal(alg)
+	if err != nil {
+		t.Fatalf("Expected no marshaling error but got: %s", err)
+	} else if expectEmptyFlagMapperPCAJSON != string(b) {
```

Comment:
I would do `string(b) != expectEmptyFlagMapperPCAJSON` instead.
This is no different from `err != nil` a couple of lines above.